module3/lessons/Week2/FeatureTesting.md

```diff
@@ -0,0 +1,327 @@
+
+1. Add a new Project to your Solution - call the project `MvcMovie.FeatureTests`.
+2. Create a Project Reference from your test project to the `MvcMovie` project.
+3. Install the following packages:
+    * Microsoft.Extentions.Hosting
+    * Microsoft.AspNetCore.Hosting
+    * Microsoft.AspNetCore.Mvc.Testing (version 6.0.16)
+4. Add a class called `Program.cs` - this is where we are going to create our test server! Add the code below to this file:
+
+```c#
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using MvcMovie.Controllers;
+
+namespace MvcMovie.FeatureTests
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            CreateHostBuilder(args).Build().Run();
+        }
+
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
+            Host.CreateDefaultBuilder(args)
+                .ConfigureWebHostDefaults(webBuilder =>
+                {
+                    webBuilder.ConfigureServices(services =>
+                    {
+                        services.AddControllersWithViews()
+                            .AddApplicationPart(typeof(HomeController).Assembly);
+                    });
+
+                    webBuilder.Configure(app =>
+                    {
+                        app.UseRouting();
+                        app.UseEndpoints(endpoints =>
+                        {
+                            endpoints.MapControllerRoute(
+                                name: "default",
+                                pattern: "{controller=Home}/{action=Index}/{id?}");
+                        });
+                    });
+                });
+    }
+}
+```
+
+5. Right-click on the test project and select `Properties`
+6. Scroll Down to `Startup object` and select `McvMovie.FeatureTests.Program`
+
+<section class="call-to-action">
+    <p>With a Partner - What similarities and differences do you see between this <code>Program.cs</code> and the <code>Program.cs</code> in the MvcMovie project?</p>
+</section>
+
+## Testing Our Welcome Page
+
+<section class="note">
+    <p>In this section, we are going to write a test for the following user story:</p>
+    <span style="white-space: pre-line">As a user
+    When I visith the home page
+    Then I see the message "Welcome"</span>
+</section>
+
+1. Update the `UnitTest1.cs` class to be `HomeControllerTests`
+2. Update the file to match the code below:
+
+```c#
+using Microsoft.AspNetCore.Mvc.Testing;
+
+namespace MvcMovie.FeatureTests
+{
+    public class HomeControllerTests : IClassFixture<WebApplicationFactory<Program>>
+    {
+        private readonly WebApplicationFactory<Program> _factory;
+
+        public HomeControllerTests(WebApplicationFactory<Program> factory)
+        {
+            _factory = factory;
+        }
+
+        [Fact]
+        public async Task Index_ShowsTheWelcomePage()
```

Comment:
This type of testing seems like a great find! Not too much set up, and we're about to do testing in a way where we think only about the user experience.